core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -577,11 +562,21 @@ public static CLICommand getCurrent() {
 
     static {
         // register option handlers that are defined
-        ClassLoaders cls = new ClassLoaders();
         Jenkins j = Jenkins.getInstanceOrNull();
         if (j != null) { // only when running on the controller
-            cls.put(j.getPluginManager().uberClassLoader);
+            // Register OptionHandlers through META-INF/services/annotations and Annotation Indexer
+            try {
+                for (Class c : Index.list(OptionHandlerExtension.class, Jenkins.get().pluginManager.uberClassLoader, Class.class)) {
+                    Type t = Types.getBaseClass(c, OptionHandler.class);
+                    CmdLineParser.registerHandler(Types.erasure(Types.getTypeArgument(t, 0)), c);
+                }
```

Comment:
Moved from lines 535-537, so please do not review this as original code authored by me.